examples/prorata/host/src/main.rs

```diff
@@ -0,0 +1,125 @@
+use ciborium;
+use clap::{Parser, Subcommand};
+use hex;
+use methods::{PRORATA_GUEST_ELF, PRORATA_GUEST_ID};
+use prorata_core::QueryResult;
+use risc0_zkvm::{
+    serde::{from_slice, to_vec},
+    Executor, ExecutorEnv,
+};
+use rust_decimal::Decimal;
+
+#[derive(Parser)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Compute an allocation for one recipient
+    Allocate {
+        /// Input file to use (CSV)
+        #[arg(short, long)]
+        input: String,
+
+        /// Output file to use (binary receipt)
+        #[arg(short, long)]
+        output: String,
+
+        /// Recipient to compute allocation for
+        #[arg(short, long)]
+        recipient: String,
+
+        /// Amount of funds to allocate
+        #[arg(short, long)]
+        amount: Decimal,
+    },
+    /// Verify an allocation read from a receipt
+    Verify {
+        /// Input file to use (binary receipt)
+        #[arg(short, long)]
+        input: String,
+    },
+}
+
+fn main() {
+    let cli = Cli::parse();
+
+    match &cli.command {
+        Commands::Allocate {
+            input,
+            output,
+            recipient,
+            amount,
+        } => allocate(input, output, recipient, amount),
+        Commands::Verify { input } => verify(input),
+    }
+}
+
+fn allocate(input: &str, output: &str, recipient: &str, amount: &Decimal) {
+    // read the file in the path specified as args.input into a vec of u8 called
+    // recipients_csv
+    let recipients_csv = std::fs::read(&input).expect("Failed to read input file");
+    let query = recipient;
```

Comment:
Why the rename/rebinding? How about just rename `recipient` parameter to `query`?